packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -1920,9 +1923,19 @@ function commitRootImpl(root, renderPriorityLevel) {
     // The first phase a "before mutation" phase. We use this phase to read the
     // state of the host tree right before we mutate it. This is where
     // getSnapshotBeforeUpdate is called.
-    prepareForCommit(root.containerInfo);
+    const activeInstanceFiber = prepareForCommit(root.containerInfo) || null;
+    let shouldFireAfterActiveInstanceBlur = false;
+
     nextEffect = firstEffect;
     do {
+      if (
+        !shouldFireAfterActiveInstanceBlur &&
+        activeInstanceFiber !== null &&
+        isFiberHiddenOrDeletedAndContains(nextEffect, activeInstanceFiber)
```

Comment:
This puts a traversal inside another traversal. Have we measured the impact of this?

Reply:
We have traversal within traversal before when we used to do these checks in `hideInstance`, so the impact is roughtly equivalent. Like I mentioned, we could maybe avoid this by putting all the current effects in a `Set` or an array, but that seems more involved and error prone.

Reply:
As @sebmarkbage mentioned last night:

> We could check if an effect is a deletion in the snapshot phase, and if it is, backtrack the active fiber's return pointer to see if we find the same fiber on three. It's not great since it'll make many deletions slower when there is at least one focused element. But it's faster for commits with no deletions, the same for one deletion and linearly worse for many deletions.

Reply:
I think his comment implies you'd only do this for deletions. But the code not only backtracks for deletions, but for every fiber. Because it does two checks (the other one being "is in hidden tree"). That second check always traverses upwards.

Reply:
>We have traversal within traversal before when we used to do these checks in hideInstance, so the impact is roughtly equivalent. 

I don't understand this. `hideInstance` is only called for things we're actually hiding. It fires very rarely. Like once a second or something.

Whereas this traversal is embedded into the hottest path — effect traversal. There can be hundreds of effect for every commit, which itself may happen dozen of times per second.

Reply:
We no longer need to do this extra traversal as I moved the code that handles this into a different place `commitBeforeMutationEffects`.

Reply:
> I don't understand this.

Do you understand this now, given your other comment? This logic was more complex before because of a subtle bug that I fixed – but I didn't remove this complexity from this location (it was the weekend and I got told off for working haha). I also rebased master and fixed the conflicts.